src/dataflow/src/source/kafka.rs

```diff
@@ -390,7 +402,28 @@ impl KafkaSourceReader {
                     old: self.last_stats.take(),
                     new: Some(stats.clone()),
                 });
-                self.last_stats = Some(stats);
+                self.last_stats = Some(stats.clone());
+            }
+            if let Some(obj) = stats.as_ref().to_serde_json().as_object() {
```

Comment:
Ah, you can use the `Deserialize` implementation on https://docs.rs/rdkafka/latest/rdkafka/statistics/struct.Statistics.html to get a properly structured type, rather than having to roll the decoding yourself!

Reply:
I thought about it but since we only needed a single element it seemed like it might be cheaper/better to just fish the single key out of the json encoded version, but it will definitely be more readable and less fragile to do it with that struct.

Reply:
>I thought about it but since we only needed a single element it seemed like it might be cheaper/better to just fish the single key out of the json encoded version, but it will definitely be more readable and less fragile to do it with that struct.

Ah, yeah, unfortunately that's not what `to_serde_json()` does; that'll allocate a `serde_json::Value` struct in one shot! So if you're going to call `to_serde_json()`, you might as well just use `Statistics`.

Reply:
If you wanted to do the fast thing, you'd need to copy/paste the parts of `Statistics` that you want into new structs...

```json
#[derive(Deserialize, Debug)]
pub struct Statistics {
    pub topics: HashMap<String, Topic>,
}

#[derive(Deserialize, Debug)]
pub struct Topic {
    pub topic: String,
    pub partitions: HashMap<i32, Partition>,
}

#[derive(Deserialize, Debug, Default)]
pub struct Partition {
    pub partition: i32,
    pub hi_offset: i64,
}

```

...and then deserialize into _that_. Which, totally reasonable if this were hot, but these stats callbacks are so infrequent that it's probably not worth it!

Reply:
I knew it would build the whole serde_json::Value, but I thought there was some laziness to how it did the expensive type conversion stuff but it seems like I was incorrect.